backend/service/topology/cache.go

```diff
@@ -66,3 +69,61 @@ func convertLockIdToAdvisoryLockId(lockID string) uint32 {
 	x := sha256.New().Sum([]byte(lockID))
 	return binary.BigEndian.Uint32(x)
 }
+
+// TODO (mcutalo): make this private once its in use
+func (c *client) SetCache(ctx context.Context, obj *topologyv1.Resource) error {
+	const upsertQuery = `
+		INSERT INTO topology_cache (id, resolver_type_url, data, metadata)
+		VALUES ($1, $2, $3, $4)
+		ON CONFLICT (id) DO UPDATE SET
+			id = EXCLUDED.id,
```

Comment:
nit: this line is unnecessary since the id will not change. could be minor implication in perf, i really don't know much about postgres perf, probably would realize that it's not changed thus no impact.